bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Display.java

```diff
@@ -3425,7 +3425,17 @@ long msgFilterProc (long code, long wParam, long lParam) {
 				MSG msg = new MSG ();
 				int flags = OS.PM_NOREMOVE | OS.PM_NOYIELD | OS.PM_QS_INPUT | OS.PM_QS_POSTMESSAGE;
 				if (!OS.PeekMessage (msg, 0, 0, 0, flags)) {
+					// Windows hooks will inherit the thread DPI awareness from
+					// the process. Whatever DPI awareness was set before on
+					// the thread will be overwritten before the hook is called.
+					// This requires to reset the thread DPi awareness to make
+					// sure, all UI updates caused by this will be executed
+					// with the correct DPI awareness
+					long previousDPIAwareness = refreshDPIAwareness();
 					if (runAsyncMessages (false)) wakeThread ();
+					if (previousDPIAwareness > 0) {
+						OS.SetThreadDpiAwarenessContext(previousDPIAwareness);
+					}
```

Comment:
What about extracting the whole set/reset logic into a separate method, so that this call looks something like this:
```suggestion
					runWithProperDPIAwareness(() -> {
						if (runAsyncMessages(false)) wakeThread();
					});
```
and `refreshDPIAwareness()` is extended to `runWithProperDPIAwareness(Runnable operation)`

Reply:
Adapted it accordingly